src/components/molecules/ChatList/ChatList.tsx

```diff
@@ -0,0 +1,140 @@
+import React, { useCallback, useState } from "react";
+import { Modal } from "react-bootstrap";
+
+import { User } from "types/User";
+
+import { WithId } from "utils/id";
+
+import { useSelector } from "hooks/useSelector";
+
+import {
+  PrivateChatMessage,
+  RestrictedChatMessage,
+} from "components/context/ChatContext";
+import UserProfileModal from "components/organisms/UserProfileModal";
+
+import "./ChatList.scss";
+import { ChatMessage } from "./ChatMessage";
+
+interface ChatListProps {
+  messages: WithId<RestrictedChatMessage | PrivateChatMessage>[];
+  emptyListMessage?: string;
+  allowDelete?: boolean;
+  deleteMessage: (id: string) => Promise<void>;
+}
+
+const ChatList: React.FC<ChatListProps> = ({
+  messages,
+  allowDelete,
+  emptyListMessage,
+  deleteMessage,
+}) => {
+  const usersById = useSelector((state) => state.firestore.data.chatUsers);
+  const [selectedUserProfile, setSelectedUserProfile] = useState<
+    WithId<User>
+  >();
+  const [showDeleteModal, setShowDeleteModal] = useState(false);
+  const [deleting, setDeleting] = useState(false);
+  const [error, setError] = useState<string | null>();
+  const [messageToDelete, setMessageToDelete] = useState<
+    WithId<RestrictedChatMessage | PrivateChatMessage>
+  >();
+
+  const showUserProfile = useCallback(
+    (message) => {
+      if (!usersById) {
+        return;
+      }
+      setSelectedUserProfile({
+        ...usersById[message.from],
+        id: message.from,
+      });
+    },
+    [usersById]
+  );
+
+  const toggleDeleteModal = useCallback((message) => {
+    setMessageToDelete(message);
+    setShowDeleteModal(true);
+  }, []);
+
+  const confirmDelete = useCallback(async () => {
+    if (messageToDelete) {
+      setDeleting(true);
+      try {
+        await deleteMessage(messageToDelete.id);
+        setShowDeleteModal(false);
+      } catch (e) {
+        setError(e.toString());
+      } finally {
+        setDeleting(false);
+      }
+    }
+  }, [deleteMessage, messageToDelete]);
+
+  const cancelDelete = useCallback(() => {
+    setError(undefined);
+    setShowDeleteModal(false);
+  }, []);
+
+  const hideUserProfile = useCallback(
+    () => setSelectedUserProfile(undefined),
+    []
+  );
```

Comment:
[NB, nit, stylistic] I tend to try and keep all of the similar actions together. So in this case: `showUserProfile`, `hideUserProfile` and the `useState` that they wrap all relate, so I would have them close together.

No need to change for this PR though.